src/buffer.rs

```diff
@@ -495,7 +495,7 @@ impl<T: Element> PyBuffer<T> {
 
         err::error_on_minusone(py, unsafe {
             ffi::PyBuffer_ToContiguous(
-                target.as_ptr() as *mut raw::c_void,
+                target.as_mut_ptr() as *mut raw::c_void,
```

Comment:
I think `.cast()` is generally recommended these days to avoid `const` -> `mut` conversions.

```suggestion
                target.as_mut_ptr().cast(),
```

Reply:
(I'll change this to use `cast`, but the fix was the function taking `target: &mut T` but not using it mutably.)